src/System.ObjectModel/src/System/Collections/ObjectModel/ObservableCollection.cs

```diff
@@ -349,22 +356,19 @@ private void OnCollectionReset()
 
         #region Private Types
 
-        // this class helps prevent reentrant calls
-        private class SimpleMonitor : IDisposable
+        // Lazily allocated and returned only when a subclass calls BlockReentrancy().
+        // Calls MonitorExit when Dispose is called.
+        private sealed class MonitorExitDisposable : IDisposable
```

Comment:
As long as we're changing this, could we pick a name that doesn't involve "Monitor"?  It sounds too much like locking is happening here with Monitor, when it's of course not.